openqaoa/optimizers/pennylane/pennylane_optimizers/adagrad.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Some lines of code have been modified from the original PennyLane source code
```

Comment:
I put in each modified file a line like this, following what says here https://www.apache.org/licenses/LICENSE-2.0, that "any modified files to carry prominent notices stating that You changed the files". Not sure if that is enough. Also, I put a comment in each modified line.

Reply:
Ok, how about:

# NOTICE #
# Lines modified by Entropica Labs will bear the comment # changed from the original PennyLane code